plugins/catalog-backend/src/ingestion/CatalogRules.ts

```diff
@@ -147,10 +147,10 @@ export class CatalogRulesEnforcer {
     }
 
     for (const matcher of matchers) {
-      if (matcher.type !== location.type) {
+      if (matcher.type !== location?.type) {
```

Comment:
yeah it's a bit extreme to expect nullish direct inputs, but while I was at it ...